### PR TITLE
provider/google: Google Cloud content-based load balancing example.

### PR DESCRIPTION
**Pull request by rileykarson**
Add a new Google Cloud example based on the  [content-based load balancing tutorial](https://cloud.google.com/compute/docs/load-balancing/http/content-based-example).

### CODE REVIEW

**Comment by rileykarson** [comment]
@danawillow

File: examples/google-content-based-load-balancing/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Run with a command like this:
+
+```
+terraform apply \
+	-var="region=us-central1" \
+	-var="region_zone=us-central1-f" \
+	-var="project_name=my-project-id-123" \
+	-var="credentials_file_path=~/.gcloud/Terraform.json" \
+	-var="public_key_path=~/.ssh/gcloud_id_rsa.pub" \
```
Comment (danawillow) [inline examples/google-content-based-load-balancing/README.md]: These last two variables aren't referenced from your tf config so they can be removed

File: examples/google-content-based-load-balancing/main.tf [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  network = "default"
+
+  allow {
+    protocol = "tcp"
+    ports = ["80"]
+  }
+
+  source_ranges = ["130.211.0.0/22", "35.191.0.0/16"]
+  target_tags = ["http-tag"]
+}
```
Comment (danawillow) [inline examples/google-content-based-load-balancing/main.tf]: nit: may as well put a newline here

File: examples/google-content-based-load-balancing/terraform.tfvars.example [hunk 3]
```diff
@@ -0,0 +1,5 @@
+region = "us-central1"
+region_zone = "us-central1-b"
+region_zone2 = "us-central1-c"
```
Comment (danawillow) [inline examples/google-content-based-load-balancing/terraform.tfvars.example]: `region_zone2` is unused

File: examples/google-content-based-load-balancing/variables.tf [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+variable "project_name" {
+  description = "The ID of the Google Cloud project"
+}
+
+variable "credentials_file_path" {
+  description = "Path to the JSON file used to describe your account credentials"
+  default     = "~/.gcloud/Terraform.json"
+}
+
+variable "public_key_path" {
```
Comment (danawillow) [inline examples/google-content-based-load-balancing/variables.tf]: Looks like these last 6 vars aren't referenced in the config

File: examples/google-content-based-load-balancing/main.tf (hunk 2 above)
Comment (rileykarson) [inline examples/google-content-based-load-balancing/main.tf]: Done.

**Pull request merged**